src/BankAccountApp/Checking.java

```diff
@@ -0,0 +1,77 @@
+package BankAccountApp;
+
+import java.util.concurrent.ThreadLocalRandom;
+
+public class Checking extends Account {
+
+    // list properties specific to a checking account
+    // debit card with pin, account number beginning with 2
+    private long debitCardNumber;
+    private int debitCardPIN;
+
+    // Constructor to init checking acct properties
+    public Checking(String name, String sSN, double initDeposit) {
+        super(name, sSN, initDeposit);
+        accountNumber = "2" + accountNumber;
+        this.debitCardNumber = setDebitCardNumber();
+        this.debitCardPIN = setDebitCardPIN();
+        this.rate = setRate();
+    }
+
+    @Override
+    public double setRate() {
+        rate = (getBaseRate() * .15);   /// why is it rounding to 0???
```

Comment:
If `rate` was an `int`, this would truncate down to 0. By default, if a number cannot fit into the bounds of it's type, it's "truncated", where the trailing digits are removed. In the case of 0.1785, .1785 cannot be represented a whole number (int) and thus the value is truncated to 0. If it was 2.1785, it would truncate to 2.